main/src/com/google/refine/browsing/Engine.java

```diff
@@ -91,7 +91,7 @@ public void setMode(Mode mode) {
         _mode = mode;
     }
 
-    public FilteredRows getAllRows() {
+    public synchronized FilteredRows getAllRows() {
```

Comment:
If look at the RowVisitor interface, the getAllRows method is NOT read only which sounds like. so  it's also necessary to protect the methods since there is client such as python-client can access it in different ways